mongodb/configure.sls

```diff
@@ -0,0 +1,79 @@
+{% from "mongodb/map.jinja" import mongodb with context %}
+
+copy_mongodb_key_file:
+  file.managed:
+    - name: {{ mongodb.cluster_key_file }}
+    - contents: "{{ salt.pillar.get('mongodb:cluster_key') }}"
+    - owner: mongodb
+    - group: mongodb
+    - mode: 0600
+    - require:
+      - pkg: install_packages
+
+stop_mongodb_service:
+  service.dead:
+    - running: True
+    - name: mongodb
+    - enable: True
+    - require:
+      - pkg: install_packages
```

Comment:
After the config is placed, the repset then needs to be generated *after* a restart completes. I tried to get this to work using `watch` and `watch_in, but eventually gave up as I had trouble getting the restart to happen **before** continuing with the repset initialization. Instead, it kept executing after everything finished, as if ignoring all of the `require_in`s I had defined in `start_mongodb_service`.